vector3d.h

```diff
@@ -0,0 +1,83 @@
+//3d vector class for Rendera, Yashas M Salian
+
+#include<math.h>
+
+class vec3
+{
+private:
+    float x, y, z;    //Components of vector are kept private
+public:
+    void xinput(float xin)    //functions used to input x, y, z
+    {
+        x = xin;
+    }   
+    
+    void yinput(float yin)
+    {
+        y = yin;
+    } 
+    
+    void zinput(float zin)
+    {
+        z = zin;
+    }
+    
+    float getx()    //functions used to obtain the stored values of x, y, z
+    {
+        return x;
+    }
+    
+    float gety()
+    {
+        return y;
+    }
+    
+    float getz()
+    {
+        return z;
+    }
+    
+    vec3 unitvect(vec3 vect)    //used to obtain unit vector in the direction of the vector vect
+    {
+        float mag;
+        vec3 unit;
```

Comment:
Same thing here. You should just construct a new vector with the new values instead of declaring and then manually setting it component by component.